### ISSUE TITLE
Exo Player 2.3 droppedOutputBufferCount get increased during dash playbck

### ISSUE THREAD

**Issue by dhanapal20** [google/ExoPlayer#3139]
### Issue description
While playing dash vod/live content using ExoPlayer 2.3, we observd "droppedOutputBufferCount" keeps incrementing (this is seen only for video and not for audio). Because of the incrementing value induces video stuttering.

### Reproduction steps
This is seen always with ExoPlayer 2.3 on all devices [android mobile (all major vendors) and android STB/TV ].

### Link to test content
Provide a link to media that reproduces the issue. If you don't wish to post it
publicly, please submit the issue, then email the link to
dev.exoplayer@gmail.com including the issue number in the subject line.

### Version of ExoPlayer being used
Exo Player 2.3

### Device(s) and version(s) of Android being used
This is seen always with ExoPlayer 2.3 on all devices [android mobile (all major vendors) and android STB/TV ].

We have cross verified the exo player code and observed that  droppedOutputBufferCount get incresed by checking bufferPresentationTimeUs and elapsedRealtimeUs difference is 30 ms
So could you kindly advice the below cases narrow down the issue
   1) How the player calculate bufferPresentationTimeUs and elapsedRealtimeUs 
   2) On what basis this 30 ms allocated as buffer time
   3) Is there any dependency in Dash manifest field with respect to this dropped buffer count or related 
       to the way in the raw video get packaged

Attachment
We have attached the problematic sample dash manifest having single video segment. if this is played the db count is increased to  8 within 2 seconds 
[vod_o_forGithub.zip](https://github.com/google/ExoPlayer/files/1200252/vod_o_forGithub.zip)

**Comment by ojw28** [comment]
It looks like the presentation timestamps of the frames in the video file are incorrect. They should be increasingly smoothly by 40000us each frame. The actual values (after decoder re-ordering) are:
```
34364096000
34364136000
34364176000
34364216000
34364256000
34364296000
34364336000
34364376000
34364416000
34364456000
34364496000
34364536000
34364576000
34364616000
34364227000 <-- Incorrect. Should be 34364656000
34364696000
34364736000
34364776000
34364387000 <-- Incorrect. Should be 34364816000
34364856000
34364896000
34364936000
34364547000 <-- Incorrect. Etc.
34365016000
34365056000
34365096000
34364707000 <-- Incorrect.
34365176000
34365216000
34365256000
34364867000 <-- Incorrect.
34365336000
34365376000
34365416000
34365027000 <-- Incorrect.
34365496000
34365536000
34365576000
34365187000 <-- Incorrect.
34365656000
34365696000
34365736000
34365347000 <-- Incorrect.
34365816000
34365856000
34365896000
34365936000
34365976000
34366016000
34366056000
```
Note there are 8 frames (indicated above) where the presentation timestamp appears to jump backward in time by a significant amount. These frames are deemed as being late and so are dropped. The media is bad and should be fixed.

**Comment by dhanapal20** [comment]
Hi
Thanks for your detailed analysis and timely response
But we have cross verified the media content and its presentation timestamp seems correct. (verified with MPEG2 TS format source content PES and its converted into iso bmff (mp4) video format with respect to monotonic clock resolution of 100 nano ticks time scale (10000000))

We upscaling 90 KHz PTS/DTS time reference to 100 nano ticks (10000000) time reference on MPEG2 TS to mp4 video conversion. Also we manually downscaled from 100 nano ticks to 90 KHz clock resolution and get back the original PTS/DTS values.

I have attached the mp4 video file analysis report with respective duration and Composition time offset values and all segments contains 400000 nano ticks for every frame. The same MPEG2 TS content playback will not induce dropped buffer count on same exo player.

So you please kindly go through the mp4 segment file with respect to this nano ticks time scale and confirm us whether this time scale will induce any issue on player presentation time stamp calculation (may be data wrap on case, etc..)
![mp4_sample](https://user-images.githubusercontent.com/29534307/28998257-f09bb882-7a43-11e7-8b69-9f8460e44c03.png)

**Comment by ojw28** [comment]
Oops! Took another look at this and it seems we overflow when parsing sample composition time offsets from the trun, [here](https://github.com/google/ExoPlayer/blob/r2.4.4/library/core/src/main/java/com/google/android/exoplayer2/extractor/mp4/FragmentedMp4Extractor.java#L814), which is why the presentation timestamps end up looking incorrect. Changing `1000` to `1000L` on that line fixes the problem. Using `Util.scaleLargeTimestamp` is possibly better. We'll push a fix on Monday and get it into the 2.5.0 release.

Aside: I'm guessing your choice of timescale is unusually large, given we haven't seen this reported before.

**Comment by dhanapal20** [comment]
Hi
Thank you very much for your analysis feedback and timely response
